src/Esprima/ArrayList.cs

```diff
@@ -370,13 +370,18 @@ public void TrimExcess(int threshold = MinAllocatedCount)
     /// </remarks>
     public Span<T> AsSpan() => new Span<T>(_items, 0, _count);
 
+    /// <remarks>
+    /// Items should not be added or removed from the <see cref="ArrayList{T}"/> while the returned <see cref="ReadOnlySpan{T}"/> is in use!
+    /// </remarks>
+    public ReadOnlySpan<T> AsReadOnlySpan() => new ReadOnlySpan<T>(_items, 0, _count);
```

Comment:
> this might allow automatic conversion

Hmm, while it may be a good idea in general, in this case I don't think so because the underlying list can be mutated as pointed out in the XML doc. It's preferable that the consumer of this API encounters this message, which they won't if we allow implicit conversion.

> not sure if would be good to sprinkle some MethodImpl(MethodImplOptions.AggressiveInlining) to As*Span methods too, just in case

This is a good idea, however. At least, it can't hurt.

Reply:
> > this might allow automatic conversion
> 
> Hmm, while it may be a good idea in general, in this case I don't think so because the underlying list can be mutated as pointed out in the XML doc. It's preferable that the consumer of this API encounters this message, which they won't if we allow implicit conversion.

A very good point. Whilst the type is internal, anyone could do that mistake. I guess my idea would still apply for `NodeList` as that one is read-only 🤔

Reply:
Yep, `NodeList` looks like a better candidate for your idea, however I don't think we'd gain much even in this case because when `NodeList`s are accessed, they are mostly enumerated in `for` or `foreach` loops. E.g.:

```cs
foreach (var node in nodeList)
{
   /* ... */
}
```

Even though we define an implicit conversion, the standard enumerator will be used instead of the ROS enumerator. For that you'd need to type out the explicit type cast to ROS, create a ROS variable before the loop, etc. Which looks more cumbersome than just snapping an `.AsSpan()` to the end of `nodeList`...